src/v2/auth.rs

```diff
@@ -143,47 +149,65 @@ impl Client {
         Box::new(auth)
     }
 
+    /// Ensure the client is authenticated with the registry.
+    pub fn ensure_auth(&self, token: Option<&str>) -> FutureBool {
```

Comment:
I think this is unused at the moment and can be dropped. If you are planning to use it in library consumer, I think it makes more sense to take ownership of `self` and return a `Future<Item = Self>`.